sqlitecluster/SQLiteNode.cpp

```diff
@@ -2580,6 +2541,77 @@ void SQLiteNode::_recvSynchronize(Peer* peer, const SData& message) {
         throw "commits remaining at end";
 }
 
+bool SQLiteNode::peerClosenessSort(const Peer* lhs, const Peer* rhs)
+{
+    // Null pointers are never fast/close.
+    if (!lhs) {
+        return false;
+    }
+    if (!rhs) {
+        return true;
+    }
+
+    // If two peers have the same latency, then the one with the higher CommitCount is closest. This case is most
+    // likely in the event that we have no latency data for either peer.
+    if (lhs->latency == rhs->latency) {
+        return SToInt((*lhs)["CommitCount"]) > SToInt((*rhs)["CommitCount"]);
+    }
+
+    // If only one peer has 0 latency, the other is faster/closer.
+    if (!lhs->latency) {
+        return false;
+    }
+    if (!rhs->latency) {
+        return true;
+    }
+
+    // Both peers have latency data, and neither are 0.
+    return lhs->latency < rhs->latency;
+}
+
+SQLiteNode::Peer* SQLiteNode::updateSyncPeer()
+{
+    // Sets are stored in sorted order. This set uses peerClosenessSort as its comparator function.
+    set<Peer*, decltype(&peerClosenessSort)> latencies {peerClosenessSort};
```

Comment:
Also, why bother actually sorting everything -- that's a lot more expensive than just finding the fastest (searching is O(N), but sorting is much higher than that).  This code seems like it could be a bit faster, but a *lot* more readable, if you just did a linear sweep over this list.